src/core/src/pattern/op/wrap_type.cpp

```diff
@@ -12,6 +12,12 @@
 bool ov::pass::pattern::op::WrapType::match_value(Matcher* matcher,
                                                   const Output<Node>& pattern_value,
                                                   const Output<Node>& graph_value) {
+    // Always check output indices for multi-output nodes
+    if (graph_value.get_node_shared_ptr()->get_output_size() > 1 &&
+        pattern_value.get_index() != graph_value.get_index()) {
+        return false;
+    }
```

Comment:
```suggestion
    if (pattern_value.get_index() != graph_value.get_index()) {
        return false;
    }
```
For single output it should be always true and for multi the index must match. Then `graph_value.get_node_shared_ptr()->get_output_size() > 1` maybe is not requried.